### DISCUSSION TITLE
Implementing libp2p::request-response::Codec for a custom type

### DISCUSSION THREAD

**Discussion by radiohertz** [libp2p/rust-libp2p#3985] [Q&A] [answered]
Hi, i was wondering if there's any reference/example to use to implement `libp2p::request-response::Codec` on my custom type.

https://docs.rs/libp2p/latest/libp2p/request_response/trait.Codec.html

**Comment by thomaseizinger** [comment, marked as answer]
We just merged https://github.com/libp2p/rust-libp2p/pull/3952!

Does that help?
There are also other implementations in the repo. If you clone it and index it with an IDE, should be easy to find :)

**Reply by radiohertz** [reply]
> We just merged #3952 

This is fantastic, exactly what i need thank you. When will this be released on crates.io?

**Reply by thomaseizinger** [reply]
When the milestone is complete: https://github.com/libp2p/rust-libp2p/milestone/6
